remotesigning/validation.go

```diff
@@ -89,3 +94,93 @@ func CalculateWitnessHashPSBT(transaction string) (*string, error) {
 	result := hex.EncodeToString(hash)
 	return &result, nil
 }
+
+func ValidateScript(signing *SigningJob, xpub string) (bool, error) {
+	// Step 1: Derive Tx Script from extended public key
+	_, nonHardenedPath, err := SplitDerivationPath(signing.DestinationDerivationPath)
+	if err != nil {
+		return false, err
+	}
+	childPubkey, err := utils.DeriveChildPubKeyFromExistingXPub(xpub, nonHardenedPath)
+	if err != nil {
+		return false, err
+	}
+	generated_script, err := GenerateScriptFromChildPubkey(childPubkey)
+	if err != nil {
+		return false, err
+	}
+
+	// Step 2: Obtain Tx Script from Change Address (Directly from Transaction)
+	txHex := *signing.Transaction
+	rawTxBytes, err := hex.DecodeString(txHex)
+	if err != nil {
+		return false, fmt.Errorf("failed to decode transaction hex: %v", err)
+	}
+
+	var tx wire.MsgTx
+	if err := tx.Deserialize(bytes.NewReader(rawTxBytes)); err != nil {
+		return false, fmt.Errorf("failed to deserialize transaction: %v", err)
+	}
+
+	if len(tx.TxOut) < 2 {
+		// TODO: May need to modify this to validate non-withdrawal L1 transactions.
+		return false, fmt.Errorf("no change output found")
+	}
+	expected_script := tx.TxOut[1].PkScript
+
+	// Step 3: Compare the two scripts
+	if !bytes.Equal(generated_script, expected_script) {
+		return false, fmt.Errorf("scripts do not match")
+	}
+
+	return true, nil
+}
+
+func SplitDerivationPath(path string) (hardenedPath []uint32, remainingPath []uint32, err error) {
+	if !strings.HasPrefix(path, "m/") {
+		return nil, nil, fmt.Errorf("invalid derivation path: derivation path must start with 'm/'")
+	}
+
+	components := strings.Split(path[2:], "/")
+	if len(components) == 0 {
+		return nil, nil, fmt.Errorf("invalid derivation path: empty component")
+	}
+
+	// Validate no empty components
+	for _, component := range components {
+		if component == "" {
+			return nil, nil, fmt.Errorf("invalid derivation path: empty component")
+		}
+	}
+	hardenedPath = make([]uint32, 0)
+	remainingPath = make([]uint32, 0)
+
+	for _, component := range components {
+		isHardened := strings.HasSuffix(component, "'")
+		if isHardened {
+			component = component[:len(component)-1]
+		}
+
+		num, err := strconv.ParseUint(component, 10, 32)
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid path: %s", component)
+		}
+
+		if isHardened {
+			hardenedPath = append(hardenedPath, uint32(num)+0x80000000)
+		} else {
+			remainingPath = append(remainingPath, uint32(num))
+		}
+	}
+
+	return hardenedPath, remainingPath, nil
+}
+
+func GenerateScriptFromChildPubkey(child_pubkey []byte) ([]byte, error) {
```

Comment:
small nit: would name this smth like `GenerateP2WPKHFromPubkey`, since this is a specific kind of script, and it doesn't care if it's a child pubkey or other pubkey